### PR TITLE
Update client data when resetting access token

### PR DESCRIPTION
**Pull request by helfi92**
If a user decides to reset the access token, the client gets a new token but all other parameters provided in the login query (expires, description, scopes) are presently being neglected. This pull-requests updates the client prior to calling the callback.

### CODE REVIEW

File: src/views/ClientCreator/ClientCreator.js [hunk 1]
```diff
@@ -34,15 +34,32 @@ export default class ClientCreator extends React.PureComponent {
...
-      );
+    this.setState({ loading: true }, async () => {
+      try {
+        const description =
+          this.state.query.description ||
+          `Client created ${new Date()} for ${this.state.query.callback_url}`;
+        const { scope } = this.state.query;
+        const scopes = (typeof scope === 'string' ? [scope] : scope).filter(
+          s => s
+        );
```
Comment (djmitche) [inline src/views/ClientCreator/ClientCreator.js]: This looks like it could be moved to a utility function :)

Comment (eliperelman) [reply src/views/ClientCreator/ClientCreator.js]:
```js
const scopes = (Array.isArray(scope) ? scope : [scope]).filter(Boolean);
```

Comment (eliperelman) [reply src/views/ClientCreator/ClientCreator.js]: Put this in utils as `toArray`, similar to Neutrino:

https://github.com/mozilla-neutrino/neutrino-dev/blob/master/packages/neutrino/src/utils.js#L18

You can use Ramda or not, your preference. :)

Comment (helfi92) [reply src/views/ClientCreator/ClientCreator.js]: Woah, I love this!

**Pull request merged**